Prod deploy pipeline

twunderlich-grapl opened the pull request:
<!-- Thank you for your contribution to Grapl! Please take some time
to fill out this short template to help us review and understand your
PR. -->

### Which issue does this PR correspond to?
https://app.zenhub.com/workspaces/grapl-6036cbd36bacff000ef314f2/issues/grapl-security/issue-tracker/832
<!-- Please provide a link to the GitHub Issue this pull request is
meant to address. -->

### What changes does this PR make to Grapl? Why?
This adds a deploy pipeline for production, represented by the mock customer accounts.

This includes the following changes:
Adding the mock customer pulumi config files
Having the merge pipeline automatically add artifacts to the mock customer config files just like it does for staging
Creating an actual production pipeline to deploy to these accounts
<!-- Please describe the functional impact of these changes and the
rationale behind them. This will help us understand your approach to
solving the problem. -->

### How were these changes tested?
Manual deploys
<!-- Please describe how these changes were tested. There should be
sufficient detail that reviewers can replicate your testing
procedure. If the procedure is a manual one that's OK, your
description will help us work with you to get it automated. -->

Commit:
Add stack config files for mock customers

Commit:
Add mock customer stacks to rc steps

Commit:
Set up initial prod deploy pipeline

This adds a prod deploy pipeline that deploys to the mock customer
accounts

christophermaier commented on .buildkite/pipeline.deploy.yml:
```diff
@@ -0,0 +1,49 @@
+      - grapl-security/vault-env#v0.1.0:
+          secrets:
+            - PULUMI_ACCESS_TOKEN
+      - improbable-eng/metahook#v0.4.1:
+          post-command: |
+            .buildkite/scripts/grapl_testing_stack/dump_artifacts.sh
```
I don't think this plugin is necessary here.

twunderlich-grapl replied:
If it turns out that we need these logs, this is easy enough to add back in. And frankly they're probably available at a different artifact path anyway. Removing

christophermaier replied:
These logs are generated from our integration / e2e tests, though, not Pulumi.

christophermaier commented on .buildkite/pipeline.deploy.yml:
```diff
@@ -0,0 +1,49 @@
+          stack: grapl/customer-mock1
+    agents:
+      queue: "pulumi-customer-mock1"
+    artifact_paths:
+      - "test_artifacts/**/*"
```
This `artifact_paths` stanza isn't needed either.

christophermaier commented on .buildkite/pipeline.deploy.yml:
```diff
@@ -0,0 +1,49 @@
+      - grapl-security/pulumi#v0.1.2:
+          command: update
+          project_dir: pulumi/grapl
+          stack: grapl/customer-mock1
```
This should be `grapl/customer-mock2`

christophermaier reviewed (approved):
Looks good, pending some minor tweaks.

I appreciate the breakdown of the commits, too 👍

Commit:
Remove unnecessary plugin for deploy pipeline

The dump artifacts script is primarily used for getting all the Nomad
container logsprimarily for debugging failed tests. However, this is
just a pulumi deploy so there aren't multiple logs we should be
collecting, just one log that goes to standard out.

Additionally this has a fix for a copy paste error, where the stack
wasn't updated. Oh well, that's what code review is for
